### DISCUSSION TITLE
What kind of dataset can use the louvain of graphscope?

### DISCUSSION THREAD

**Discussion by neblen** [alibaba/GraphScope#379] [Q&A] [answered]
I use the example dataset of graphscope and use the project method for getting a subgraph-> 
simple_graph = g.project(vertices={"person": []}, edges={"knows": ["dist"]})
but it can't work with  the louvain of graphscope.
1.code is following
import graphscope
graphscope.set_option(show_log=True)
graphscope.set_option(log_level="DEBUG")
sess = graphscope.session(addr='x.x.x.x:xx', k8s_service_type="LoadBalancer")
g = sess.g()
g = g.add_vertices("/tmp/testingdata/property/p2p-31_property_v_0", label="person")
g = g.add_edges("/tmp/testingdata/property/p2p-31_property_e_0", label="knows")
print(g.schema)
simple_graph = g.project(vertices={"person": []}, edges={"knows": ["dist"]})
print(simple_graph)
from graphscope import louvain
result_context=louvain(simple_graph)
2.erro is following
 ---------------------------------------------------------------------------
InvalidArgumentError                      Traceback (most recent call last)
<ipython-input-3-3846851254bc> in <module>
      1 from graphscope import louvain
----> 2 result_context=louvain(simple_graph)

<decorator-gen-149> in louvain(graph, min_progress, progress_tries)

/opt/conda/lib/python3.8/site-packages/graphscope/framework/app.py in project_to_simple(func, *args, **kwargs)
     46     if graph.graph_type == types_pb2.ARROW_PROPERTY:
     47         graph = graph._project_to_simple()
---> 48     return func(graph, *args[1:], **kwargs)
     49 
     50 

<decorator-gen-148> in louvain(graph, min_progress, progress_tries)

/opt/conda/lib/python3.8/site-packages/graphscope/framework/app.py in _not_compatible_for(not_compatible_for_func, *args, **kwargs)
     98             )
     99         else:
--> 100             return not_compatible_for_func(*args, **kwargs)
    101 
    102     return _not_compatible_for

/opt/conda/lib/python3.8/site-packages/graphscope/analytical/app/louvain.py in louvain(graph, min_progress, progress_tries)
     65     """
     66     if graph.is_directed():
---> 67         raise InvalidArgumentError("Louvain not support directed graph.")
     68     return AppAssets(algo="louvain")(graph, min_progress, progress_tries)

InvalidArgumentError: 'Louvain not support directed graph.'

**Comment by sighingnow** [comment]
I think it is an expected error. Moved to discussion.

**Comment by sighingnow** [comment]
@siyuan0322 Any further comment/doc on louvain algorithm?

**Comment by acezen** [comment, marked as answer]
the louvain algorithm now only support undirected graph, you can use louvain like:
```python
g = sess.g(directed=False)
g = g.add_vertices("/tmp/testingdata/property/p2p-31_property_v_0", label="person")
g = g.add_edges("/tmp/testingdata/property/p2p-31_property_e_0", label="knows")
simple_graph = g.project(vertices={"person": []}, edges={"knows": ["dist"]})
from graphscope import louvain
result_context=louvain(simple_graph)
```

**Reply by neblen** [reply]
thank you so much
this example can work in notebook from ali's jupyter notebook.
If person want to work in personal cluster ,you need to give your notebook more resources.
